packages/nuxt/src/app/composables/asyncData.ts

```diff
@@ -200,6 +200,29 @@ export function refreshNuxtData (keys?: string | string[]): Promise<void> {
   return useNuxtApp().callHook('app:data:refresh', _keys)
 }
 
+export function clearNuxtCache ({ keys }: {keys: string | string[]}): void {
+  if (process.server) {
+    return
+  }
+
+  const _keys = keys ? Array.isArray(keys) ? keys : [keys] : undefined
+  if (_keys === undefined) {
+    return
+  }
+
+  const nuxt = useNuxtApp()
+
+  for (const key of _keys) {
+    if (nuxt.payload.data[key] !== undefined) {
+      nuxt.payload.data[key] = undefined
+    }
+
+    if (nuxt.payload._errors[key] !== undefined) {
+      nuxt.payload._errors[key] = undefined
+    }
+  }
```

Comment:
Why do you have to iterate through all the keys?
Can't you just do 
```suggestion
  nuxt.payload.data = undefined
  nuxt.payload._errors = undefined
```

Reply:
Or am I missing something stupid here?

Reply:
Because we should not delete the entire existing cache, but only some parts by keys